extensions/positron-reticulate/src/extension.ts

```diff
@@ -0,0 +1,475 @@
+	public onDidChangeRuntimeState: vscode.Event<positron.RuntimeState>;
+
+	/** An object that emits an event when the user's session ends and the runtime exits */
+	public onDidEndSession: vscode.Event<positron.LanguageRuntimeExit>;
+
+	constructor(
+		readonly rSession: positron.LanguageRuntimeSession,
+		runtimeMetadata: positron.LanguageRuntimeMetadata,
+		sessionMetadata: positron.RuntimeSessionMetadata,
+		sessionType: ReticulateRuntimeSessionType,
+	) {
+		// When the kernelSpec is undefined, the PythonRuntimeSession
+		// will perform a restore session.
+		let kernelSpec: JupyterKernelSpec | undefined = undefined;
+		if (sessionType === ReticulateRuntimeSessionType.Create) {
+			kernelSpec = {
+				'argv': [],
+				'display_name': "Reticulate Python Session", // eslint-disable-line
+				'language': 'Python',
+				'env': {},
+				'startKernel': async (session, kernel) => {
+					try {
+						await this.startKernel(session, kernel);
+					} catch (err: any) {
+						// Any error when trying to start kernel is caught and we send an error
+						// notification.
+						vscode.window.showErrorMessage(vscode.l10n.t(
+							'Failed to initialize and connect to the Reticulate Python session: {0}',
+							err.message
+						));
+						throw err;
+					}
+				},
+			};
+		}
+
+		const api = vscode.extensions.getExtension('ms-python.python')?.exports;
+		this.pythonSession = api.positron.createPythonRuntimeSession(
+			runtimeMetadata,
+			sessionMetadata,
+			kernelSpec
+		);
+
+		this.onDidReceiveRuntimeMessage = this.pythonSession.onDidReceiveRuntimeMessage;
+		this.onDidChangeRuntimeState = this.pythonSession.onDidChangeRuntimeState;
+		this.onDidEndSession = this.pythonSession.onDidEndSession;
+	}
+
+	// A function that sstarts a kernel and then connects to it.
+	async startKernel(session: JupyterSession, kernel: JupyterKernel) {
+		kernel.log('Starting the reticulate session!');
+
+		// Store a reference to the kernel, so the session can log, reconnect, etc.
+		this.kernel = kernel;
+
+		const connnectionFile = session.state.connectionFile;
+		const logFile = session.state.logFile;
+		const profileFile = session.state.profileFile;
+		const logLevel = 'debug';
+
+		const kernelPath = `${__dirname}/../../positron-python/python_files/positron/positron_language_server.py`;
+		const code = `reticulate:::py_run_file_on_thread(
+					file = "${kernelPath}",
+					args = c(
+						"-f", "${connnectionFile}",
+						"--logfile", "${logFile}",
+						"--loglevel", "${logLevel}",
+						"--session-mode", "console"
+					)
+				)`;
+
+		if (!this.rSession) {
+			kernel.log('No R session :(');
+			throw new Error('No R session to attach the Reticulate Python kernel');
+		}
+
+		// There's currently no way to discover if this execution worked or not.
+		// We just *hope* it worked and try to connect later.
+		// Connecting has a timeout of 20s before failing definitively.
+		this.rSession.execute(
```

Comment:
Is there no reply from the kernel e.g. status busy/idle messages? In a few cases in both Python and R extensions we execute code and wait for the status idle message and timeout if it isn't received.

Reply:
I'm planning to move this into a session UI method, so we can call it and wait for it's execution/ get error messages if something goes wrong.

Reply:
Should be resolved with https://github.com/posit-dev/positron/pull/4603/commits/6fa63f2eebef287a7d01d327115152131d369738